api/src/main/java/io/minio/MinioClient.java

```diff
@@ -957,6 +957,8 @@ private Request createRequest(Method method, String bucketName, String objectNam
           len = 0;
         }
 
+        // Issue #820. Work-around for okhttp3 default Accept-Encoding gzip
```

Comment:
This fix is not a workaround. Here we are overriding okhttp's default behaviour,  Refer https://square.github.io/okhttp/calls/#rewriting-requests and  https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding.

All comments in https://github.com/minio/minio-java/pull/828 are applicable to this PR as well.